noaa.go

```diff
@@ -33,90 +33,147 @@ type PointsResponse struct {
 type ForecastResponse struct {
 	// capture data from the forecast
-	Updated   string `json:"updated"`
-	Units     string `json:"units"`
-	Elevation struct {
-		Value float64 `json:"value"`
-		Units string  `json:"unitCode"`
-	} `json:"elevation"`
-	Periods []struct {
-		ID              int32   `json:"number"`
-		Name            string  `json:"name"`
-		StartTime       string  `json:"startTime"`
-		EndTime         string  `json:"endTime"`
-		IsDaytime       bool    `json:"isDaytime"`
-		Temperature     float64 `json:"temperature"`
-		TemperatureUnit string  `json:"temperatureUnit"`
-		WindSpeed       string  `json:"windSpeed"`
-		WindDirection   string  `json:"windDirection"`
-		Summary         string  `json:"shortForecast"`
-		Details         string  `json:"detailedForecast"`
-	} `json:"periods"`
-	Point *PointsResponse
+	Updated   string                    `json:"updated"`
+	Units     string                    `json:"units"`
+	Elevation *ForecastElevation        `json:"elevation"`
+	Periods   []*ForecastResponsePeriod `json:"periods"`
+	Point     *PointsResponse
+}
+
+// WeatherValueItem holds the JSON values for a weather.values[x].value.
+type WeatherValueItem struct {
+	Coverage  string `json:"coverage"`
+	Weather   string `json:"weather"`
+	Intensity string `json:"intensity"`
+}
+
+// WeatherValue holds the JSON value for a weather.values[x] value.
+type WeatherValue struct {
+	ValidTime string              `json:"validTime"` // ISO 8601 time interval, e.g. 2019-07-04T18:00:00+00:00/PT3H
+	Value     []*WeatherValueItem `json:"value"`
+}
+
+// Weather holds the JSON value for the weather object.
+type Weather struct {
+	Values []*WeatherValue `json:"values"`
+}
+
+// HazardValueItem holds a value item from a GridpointForecastResponse's hazard.values[x].value[x].
+type HazardValueItem struct {
+	Phenomenon   string `json:"phenomenon"`
+	Significance string `json:"significance"`
+	EventNumber  int32  `json:"event_number"`
+}
+
+// HazardValue holds a hazard value from a GridpointForecastResponse's hazard.values[x].
+type HazardValue struct {
+	ValidTime string             `json:"validTime"` // ISO 8601 time interval, e.g. 2019-07-04T18:00:00+00:00/PT3H
+	Value     []*HazardValueItem `json:"value"`
+}
+
+// Hazard holds a hazard item from a GridpointForecastRespones's hazards
+type Hazard struct {
+	Values []*HazardValue `json:"values"`
+}
+
+type HourlyForecastResponse struct {
+	Updated           string                          `json:"updated"`
+	Units             string                          `json:"units"`
+	ForecastGenerator string                          `json:"forecastGenerator"`
+	GeneratedAt       string                          `json:"generatedAt"`
+	UpdateTime        string                          `json:"updateTime"`
+	ValidTimes        string                          `json:"validTimes"`
+	Periods           []*ForecastResponsePeriodHourly `json:"periods"`
+	Point             *PointsResponse
```

Comment:
This is the one item that might need to stay as a pointer to a struct. I need to spend some time thinking about how the public interface to this module would be impacted if it were to be converted to a non-pointer type. (as used here and elsewhere, its in a few places that pre-date your PR.)